index.bs

```diff
@@ -4437,6 +4439,252 @@ The [=remote end subscribe steps=], with [=subscribe priority=] 10, given
+        params: InputPerformActionsParameters
+      };
+
+      InputPerformActionsParameters = {
+        context: BrowsingContext,
+        actions: [* InputSourceActions]
+      };
+
+      InputSourceActions = (
+        NoneSourceActions //
+        KeySourceActions //
+        PointerSourceActions //
+        WheelSourceActions
+      );
+
+      NoneSourceActions = {
+        type: "none",
+        actions: [* NoneSourceAction]
+      };
+
+      NoneSourceAction = PauseAction;
+
+      KeySourceActions = {
+        type: "key",
+        actions: [* KeySourceAction]
+      };
+
+      KeySourceAction = (
+        PauseAction //
+        KeyDownAction //
+        KeyUpAction
+      );
+
+      PointerSourceActions = {
+        type: "pointer",
+        ? parameters: PointerParameters,
+        actions: [* PointerSourceAction]
+      };
+
+      PointerParameters = {
+        ? pointerType: "mouse" / "pen" / "touch"
+      };
+
+      PointerSourceAction = (
+        PauseAction //
+        PointerDownAction //
+        PointerUpAction //
+        PointerMoveAction
+      );
+
+      WheelSourceActions = {
+        type: "wheel",
+        actions: [* WheelSourceAction]
+      };
+
+      WheelSourceAction = (
+        PauseAction //
+        WheelScrollAction
+      );
+
+      PauseAction = {
+        type: "pause",
+        ? duration: uint
+      };
+
+      KeyDownAction = {
+        type: "keyDown",
+        value: text
+      };
+
+      KeyUpAction = {
+        type: "keyUp",
+        value: text
+      };
+
+      PointerUpAction = {
+        type: "pointerUp",
+        button: uint,
+        PointerCommonProperties,
+      };
+
+      PointerDownAction = {
+        type: "pointerUp",
+        button: uint,
+        PointerCommonProperties
+      };
+
+      PointerMoveAction = {
+        type: "pointerMove",
+        x: uint,
+        y: uint,
+        ? duration: uint,
+        ? origin: Origin,
+        PointerCommonProperties,
+      };
+
+      WheelScrollAction = {
+        type: "scroll",
+        x: uint,
+        y: unint,
+        deltaX: int,
+        deltaY: int,
+        ? duration: uint,
+        ? origin: Origin,
+        PointerCommonProperties,
+      };
+
+      PointerCommonProperties = (
+        ? width: uint,
+        ? height: uint,
+        ? pressure: float,
+        ? tangentialPressure: float,
+        ? tiltX: int,
+        ? tiltY: int,
+        ? twist: uint,
+        ? altitudeAngle: float,
+        ? azimuthAngle: float,
+      )
+
+      Origin = "viewport" / "pointer" / ElementOrigin;
+
+      ElementOrigin = {
+        "type": "Element",
+        sharedId: SharedId
+      }
+    </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        EmptyResult
+    </pre>
+   </dd>
+</dl>
+
+A map matching the <code>ElementOrigin</code> production [=represents a web
+element=].
+
+<div algorithm>
+
+To <dfn>get a known element</a> for an <code>ElementOrigin</code>:
```

Comment:
Add a TODO here so it doesn't look like the following remote end steps are related?